modules/kafka/kafka-dest-worker.c

```diff
@@ -56,21 +58,63 @@ _format_message_and_key(KafkaDestWorker *self, LogMessage *msg)
                         self->super.seq_num, NULL, self->key);
 }
 
+static rd_kafka_topic_t *
+_calculate_topic_from_template(KafkaDestWorker *self, LogMessage *msg)
+{
+  KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
+  log_template_format(owner->topic_name, msg, &owner->template_options, LTZ_SEND, self->super.seq_num, NULL,
+                      self->topic_name_buffer);
+  rd_kafka_topic_t *topic = kafka_dd_query_insert_topic(owner, self->topic_name_buffer->str);
+
+  if (!topic)
+    {
+      /* if topic name is invalid, then topic is fallback topic which we know is defined*/
```

Comment:
It is better to make sure, we do not have any programming errors, and `g_assert()` to the fallback topic here.
That would also make this comment obsolete.